sdk/core/azure-core/src/main/java/com/azure/core/http/policy/MaxAttemptRedirectStrategy.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpMethod;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * A default implementation of {@link RedirectStrategy} that uses the provided try count, and status code
+ * to determine if request should be redirected.
+ */
+public class MaxAttemptRedirectStrategy implements RedirectStrategy {
+
+    // Based on Stamp specific redirects design doc
+    static final int MAX_REDIRECT_ATTEMPTS = 10;
+    private static final String LOCATION_HEADER_NAME = "Location";
+
+    private int maxAttempts;
+    private final String locationHeader;
+    private final Set<HttpMethod> redirectMethods;
+
+    /**
+     * Creates an instance of {@link MaxAttemptRedirectStrategy}.
+     *
+     * @param maxAttempts The max number of redirect attempts that can be made.
+     * @param locationHeader The header name containing the redirect URL.
+     * @param allowedMethods The set of {@link HttpMethod} that are allowed to be redirected.
+     *
+     * @throws NullPointerException if {@code locationHeader} is {@code null}.
```

Comment:
This is missing documentation on additional reasons why this would throw an exception.